tensorflow_addons/seq2seq/tests/basic_decoder_test.py

```diff
@@ -319,274 +319,243 @@ def test_step_with_sample_embedding_helper():
+    assert len(step_state) == 2
+    assert isinstance(step_outputs, basic_decoder.BasicDecoderOutput)
+    assert (batch_size, vocabulary_size) == step_outputs[0].get_shape()
+    assert (batch_size,) == step_outputs[1].get_shape()
+    assert (batch_size, vocabulary_size) == first_state[0].get_shape()
+    assert (batch_size, vocabulary_size) == first_state[1].get_shape()
+    assert (batch_size, vocabulary_size) == step_state[0].get_shape()
+    assert (batch_size, vocabulary_size) == step_state[1].get_shape()
+    assert (batch_size, input_depth) == step_next_inputs.get_shape()
 
-            self.evaluate(tf.compat.v1.global_variables_initializer())
-            eval_result = self.evaluate(
-                {
-                    "batch_size": batch_size_t,
-                    "first_finished": first_finished,
-                    "first_inputs": first_inputs,
-                    "first_state": first_state,
-                    "step_outputs": step_outputs,
-                    "step_state": step_state,
-                    "step_next_inputs": step_next_inputs,
-                    "step_finished": step_finished,
-                }
-            )
+    eval_result = {
+        "batch_size": batch_size_t,
+        "first_finished": first_finished,
+        "first_inputs": first_inputs,
+        "first_state": first_state,
+        "step_outputs": step_outputs,
+        "step_state": step_state,
+        "step_next_inputs": step_next_inputs,
+        "step_finished": step_finished,
+    }
 
-            self.assertAllEqual(
-                [False, False, False, False, True], eval_result["first_finished"]
-            )
-            self.assertAllEqual(
-                [False, False, False, True, True], eval_result["step_finished"]
-            )
-            sample_ids = eval_result["step_outputs"].sample_id
-            self.assertEqual(output_dtype.sample_id, sample_ids.dtype)
-            batch_where_not_sampling = np.where(sample_ids == -1)
-            batch_where_sampling = np.where(sample_ids > -1)
-            self.assertAllClose(
-                eval_result["step_next_inputs"][batch_where_sampling],
-                embeddings[sample_ids[batch_where_sampling]],
-            )
-            self.assertAllClose(
-                eval_result["step_next_inputs"][batch_where_not_sampling],
-                np.squeeze(inputs[batch_where_not_sampling, 1], axis=0),
-            )
+    np.testing.assert_equal(
+        np.asanyarray([False, False, False, False, True]),
+        eval_result["first_finished"].numpy(),
+    )
+    np.testing.assert_equal(
+        np.asanyarray([False, False, False, True, True]),
+        eval_result["step_finished"].numpy(),
+    )
+    sample_ids = eval_result["step_outputs"].sample_id
+    assert output_dtype.sample_id == sample_ids.dtype
+    batch_where_not_sampling = tf.cast(np.where(sample_ids == -1), dtype=tf.int32)
+    batch_where_sampling = tf.cast(np.where(sample_ids > -1), dtype=tf.int32)
```

Comment:
Could you explain a bit where the tf.cast come from? Are you fixing a bug or something like that?

Reply:
Thanks @gabrieldemarmiesse. This part may not be necessary, it just uses tensor for slicing. Using original slicing can't pass the testing, so come up with this way; not sure any easy way to fix it.

Reply:
Can we just quickly check the `dtype` of `sample_ids` before casting? I am asking because it shouldn't have passed the test cases earlier as well if casting was required

Reply:
@AakashKumarNain , tests were passing before because all tensors were numpy tensors or tuples. See https://github.com/tensorflow/addons/pull/1678#issuecomment-614757184